qulice-checkstyle/src/main/java/com/qulice/checkstyle/DiamondOperatorCheck.java

```diff
@@ -94,21 +94,22 @@ private static boolean isNotObjectBlock(final DetailAST node) {
     }
 
     /**
-     * Gets the return type of method or field type.
+     * Returns the first child node of a specified type, even
+     * in case when node contains as child an inner class type.
      *
      * @param node AST subtree to process.
      * @param type Type of token
      * @return Child node of specified type
      */
-    private static DetailAST getFirstChildNodeOfType(
+    private static DetailAST findFirstChildNodeOfType(
         final DetailAST node, final int type
     ) {
-        DetailAST result = node.getFirstChild();
-        if (result != null && result.getType() != type) {
-            final DetailAST child = DiamondOperatorCheck
-                .getFirstChildNodeOfType(result, type);
-            if (child == null) {
-                result = result.getNextSibling();
+        DetailAST result = node.findFirstToken(type);
+        if (result == null) {
+            final DetailAST dot = node.findFirstToken(TokenTypes.DOT);
```

Comment:
@dskalenko can you post me this tree? You can nicely visualize it in Checkstyle GUI: http://checkstyle.sourceforge.net/apidocs/com/puppycrawl/tools/checkstyle/gui/Main.html